### DISCUSSION TITLE
Enabling TLS connections with Milvus

### DISCUSSION THREAD

**Discussion by ganderaj** [milvus-io/milvus#26411] [Q&A and General discussion]
We've had deployed Milvus using [milvus-helm](https://github.com/zilliztech/milvus-helm/tree/master) on AWS EKS Cluster. We have placed "milvus" and "milvus-attu" services behind NGINX Ingress Controller and setup internal CNAME for LB endpoint.

We are now able to access Attu at attu.company.com. However, Milvus connections using `pymilvus` are timing out. Upon inspecting NGINX logs, it is learned that NGINX is unable to honor HTTP/2 over port 80 (http). 
Log message -- "PRI * HTTP/2.0" 400 157 "-" "-" "-"

We are looking at switching our traffic to HTTPS from HTTP to resolve above. We have enabled `tls` on Ingress controller. Now Attu works over secure connection (HTTPS) but Milvus (pymilvus.connection) does not.

I have gone through [Encryption in Transit](https://milvus.io/docs/tls.md) doc which is a bit confusing to me. Requesting your inputs on below --

- Do we have to create another set of certificates besides the ones enabled on our Ingress controller to access Milvus (using pymilvus) to enable encryption ?
- If yes, how do I pass new set of certificate details to Milvus ? [Modify Milvus server configurations](https://milvus.io/docs/tls.md#Modify-Milvus-server-configurations) talks about specifying paths to certs. What are these paths ? How do I inject my keys to the cluster ?

Your response is appreciated.
Thanks.

**Comment by ganderaj** [comment]
. @haorenfsa , can you help me here ?

**Comment by haorenfsa** [comment]
@ganderaj It's very likely because of the way you config your pymilvus connection method. 
Our [cloud sdk examples](https://github.com/zilliztech/cloud-vectordb-examples/blob/master/python/hello_zilliz_vectordb.py) also works for open-source Milvus. 
The key part is like below:

```python
from pymilvus import connection
connections.connect("default", uri="https://<your-milvus-address>:<port>")
```

**Comment by ganderaj** [comment]
@haorenfsa 

I have tried using "uri" as well, no luck with my setup.

Can you please comment on my questions wrt Certificates for encryption on Milvus ?

**Reply by haorenfsa** [reply]
Which version of pymilvus are you using? 2.2.11+ should be working. Or maybe you cloud provide me your client side configuration code.

**Reply by haorenfsa** [reply]
About certificates questions
no, usually you don't need to configure tls on milvus side (except GCP, their LB needs backend to enable tls). AWS or nginx Ingress supports plain text backend. You only need to config ingress' backend protocol to GRPC or HTTP2.
